synthesizer/program/src/lib.rs

```diff
@@ -966,6 +976,69 @@ impl<N: Network> ProgramCore<N> {
 
         function_contains || closure_contains || command_contains || array_size_exceeds
     }
+
+    /// Returns `true` if a program contains any string type.
+    /// Before ConsensusVersion::V12, variable-length string sampling when using them as inputs caused deployment synthesis to be inconsistent and abort with probability 63/64.
+    /// After ConsensusVersion::V12, string types are disallowed.
+    #[inline]
+    pub fn contains_string_type(&self) -> bool {
```

Comment:
Seems like i was preemptive in my approval. There are some potential missing checks - 

- [ ] Function operands
- [ ] Function inputs
- [ ] Function Outputs
- [ ] Closure operands
- [ ] Closure inputs (missing but should be accounted for by function operands)
- [ ] Closure outputs (missing but should be accounted for by closure oprands)
- [ ] Finalize operands
- [ ] Finalize inputs (missing but should be accounted for by function operands)
- [ ] Finalize outputs (missing but should be accounted for by the finalize operands)
- [ ] Structs (missing)
- [ ] Records (missing)